src/parser/hunter/survival/modules/spells/azeritetraits/BlurOfTalons.tsx

```diff
@@ -48,76 +49,69 @@ class BlurOfTalons extends Analyzer {
     this.agility = agility;
     this.blurOfTalonStacks = Array.from({ length: MAX_BLUR_STACKS + 1 }, x => []);
 
-    this.statTracker.add(SPELLS.BLUR_OF_TALONS_BUFF.id, {
-      agility,
+    options.statTracker.add(SPELLS.BLUR_OF_TALONS_BUFF.id, {
+      agility: this.agility,
     });
   }
-
-  handleStacks(event, stack = null) {
-    if (event.type === EventType.RemoveBuff || isNaN(event.stack)) { //NaN check if player is dead during on_finish
-      event.stack = 0;
-    }
-    if (event.type === EventType.ApplyBuff) {
-      event.stack = 1;
-    }
-    if (stack) {
-      event.stack = stack;
-    }
-    this.blurOfTalonStacks[this.lastBlurStack].push(event.timestamp - this.lastBlurUpdate);
-    this.lastBlurUpdate = event.timestamp;
-    this.lastBlurStack = event.stack;
-  }
-
   get blurOfTalonsTimesByStacks() {
     return this.blurOfTalonStacks;
   }
-
-  on_byPlayer_applybuff(event) {
+  get uptime() {
+    return this.selectedCombatant.getBuffUptime(SPELLS.BLUR_OF_TALONS_BUFF.id) / this.owner.fightDuration;
+  }
+  get avgAgility() {
+    const avgAgi = this.blurOfTalonStacks.reduce((sum, innerArray, outerArrayIndex) => {
+      return sum + innerArray.reduce((sum, arrVal) => sum + ((arrVal * outerArrayIndex * this.agility) / this.owner.fightDuration), 0);
+    }, 0);
+    return avgAgi;
+  }
+  handleStacks(event: RemoveBuffEvent | ApplyBuffEvent | ApplyBuffStackEvent | FightEndEvent, stack: number = 0) {
+    if (event.type === EventType.RemoveBuff) {
+      this.currentStacks = 0;
+    } else if (event.type === EventType.ApplyBuff) {
+      this.currentStacks = 1;
+    } else if (event.type === EventType.ApplyBuffStack) {
+      this.currentStacks = event.stack;
+    } else if (event.type === EventType.FightEnd) {
```

Comment:
I plan to make a separate PR pulling these handleStacks() out of their individual modules and into a core module (or at least a shared hunter module) to avoid repetition, so I'll handle it there